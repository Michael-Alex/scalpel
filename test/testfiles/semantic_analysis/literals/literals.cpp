bool bf = false;
bool bt = true;

char cc0 = 'a';
char cc1 = '\t';
char cc2 = '\7';
char cc3 = '\43';
char cc4 = '\102';
char cc5 = '\x2';
char cc6 = '\x2f';
char cc7 = '\u0060';
char cc8 = '\U00000060';
wchar_t cw0 = L'a';
wchar_t cw1 = L'\xe3f';
wchar_t cw2 = L'\u00e9';
int ci0 = 'aa';
int ci1 = '\u00e9';
int ci2 = '\U000002ff';
int ci3 = '\u00600';
int ci4 = '\u00ff0';

double fd0 = 0.0;
double fd1 = .4e5;
float ff0 = 1.f;
float ff1 = 1.0e-2f;
long double fld = 12.1e+4l;

int i = 0;
long int li = 1l;
long long int lli = 020ll;
unsigned int ui = 0x3aFu;
unsigned long int uli = 0X400ul;
unsigned long long int ulli = 50000llu;

