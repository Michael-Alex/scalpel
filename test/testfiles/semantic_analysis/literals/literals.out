{
  "id": 0,
  "variables":
  [
    {
      "id": 0,
      "name": "bf",
      "type":
      {
        "category": "fundamental type",
        "type": "bool"
      },
      "default value":
      {
        "type": "bool",
        "value": false
      }
    },
    {
      "id": 1,
      "name": "bt",
      "type":
      {
        "category": "fundamental type",
        "type": "bool"
      },
      "default value":
      {
        "type": "bool",
        "value": true
      }
    },
    {
      "id": 2,
      "name": "cc0",
      "type":
      {
        "category": "fundamental type",
        "type": "char"
      },
      "default value":
      {
        "type": "char",
        "value": 97
      }
    },
    {
      "id": 3,
      "name": "cc1",
      "type":
      {
        "category": "fundamental type",
        "type": "char"
      },
      "default value":
      {
        "type": "char",
        "value": 9
      }
    },
    {
      "id": 4,
      "name": "cc2",
      "type":
      {
        "category": "fundamental type",
        "type": "char"
      },
      "default value":
      {
        "type": "char",
        "value": 7
      }
    },
    {
      "id": 5,
      "name": "cc3",
      "type":
      {
        "category": "fundamental type",
        "type": "char"
      },
      "default value":
      {
        "type": "char",
        "value": 35
      }
    },
    {
      "id": 6,
      "name": "cc4",
      "type":
      {
        "category": "fundamental type",
        "type": "char"
      },
      "default value":
      {
        "type": "char",
        "value": 66
      }
    },
    {
      "id": 7,
      "name": "cc5",
      "type":
      {
        "category": "fundamental type",
        "type": "char"
      },
      "default value":
      {
        "type": "char",
        "value": 2
      }
    },
    {
      "id": 8,
      "name": "cc6",
      "type":
      {
        "category": "fundamental type",
        "type": "char"
      },
      "default value":
      {
        "type": "char",
        "value": 47
      }
    },
    {
      "id": 9,
      "name": "cc7",
      "type":
      {
        "category": "fundamental type",
        "type": "char"
      },
      "default value":
      {
        "type": "char",
        "value": 96
      }
    },
    {
      "id": 10,
      "name": "cc8",
      "type":
      {
        "category": "fundamental type",
        "type": "char"
      },
      "default value":
      {
        "type": "char",
        "value": 96
      }
    },
    {
      "id": 11,
      "name": "ci0",
      "type":
      {
        "category": "fundamental type",
        "type": "int"
      },
      "default value":
      {
        "type": "int",
        "value": 24929
      }
    },
    {
      "id": 12,
      "name": "ci1",
      "type":
      {
        "category": "fundamental type",
        "type": "int"
      },
      "default value":
      {
        "type": "int",
        "value": 50089
      }
    },
    {
      "id": 13,
      "name": "ci2",
      "type":
      {
        "category": "fundamental type",
        "type": "int"
      },
      "default value":
      {
        "type": "int",
        "value": 52159
      }
    },
    {
      "id": 14,
      "name": "ci3",
      "type":
      {
        "category": "fundamental type",
        "type": "int"
      },
      "default value":
      {
        "type": "int",
        "value": 24624
      }
    },
    {
      "id": 15,
      "name": "ci4",
      "type":
      {
        "category": "fundamental type",
        "type": "int"
      },
      "default value":
      {
        "type": "int",
        "value": 12828464
      }
    },
    {
      "id": 16,
      "name": "cw0",
      "type":
      {
        "category": "fundamental type",
        "type": "wchar_t"
      },
      "default value":
      {
        "type": "wchar_t",
        "value": 97
      }
    },
    {
      "id": 17,
      "name": "cw1",
      "type":
      {
        "category": "fundamental type",
        "type": "wchar_t"
      },
      "default value":
      {
        "type": "wchar_t",
        "value": 3647
      }
    },
    {
      "id": 18,
      "name": "cw2",
      "type":
      {
        "category": "fundamental type",
        "type": "wchar_t"
      },
      "default value":
      {
        "type": "wchar_t",
        "value": 233
      }
    },
    {
      "id": 19,
      "name": "fd0",
      "type":
      {
        "category": "fundamental type",
        "type": "double"
      },
      "default value":
      {
        "type": "double",
        "value": 0
      }
    },
    {
      "id": 20,
      "name": "fd1",
      "type":
      {
        "category": "fundamental type",
        "type": "double"
      },
      "default value":
      {
        "type": "double",
        "value": 400000
      }
    },
    {
      "id": 21,
      "name": "ff0",
      "type":
      {
        "category": "fundamental type",
        "type": "float"
      },
      "default value":
      {
        "type": "float",
        "value": 1
      }
    },
    {
      "id": 22,
      "name": "ff1",
      "type":
      {
        "category": "fundamental type",
        "type": "float"
      },
      "default value":
      {
        "type": "float",
        "value": 0.01
      }
    },
    {
      "id": 23,
      "name": "fld",
      "type":
      {
        "category": "fundamental type",
        "type": "long double"
      },
      "default value":
      {
        "type": "long double",
        "value": 121000
      }
    },
    {
      "id": 24,
      "name": "ii",
      "type":
      {
        "category": "fundamental type",
        "type": "int"
      },
      "default value":
      {
        "type": "int",
        "value": 0
      }
    },
    {
      "id": 25,
      "name": "ili",
      "type":
      {
        "category": "fundamental type",
        "type": "long int"
      },
      "default value":
      {
        "type": "long int",
        "value": 1
      }
    },
    {
      "id": 26,
      "name": "illi",
      "type":
      {
        "category": "fundamental type",
        "type": "long long int"
      },
      "default value":
      {
        "type": "long long int",
        "value": 16
      }
    },
    {
      "id": 27,
      "name": "iui",
      "type":
      {
        "category": "fundamental type",
        "type": "unsigned int"
      },
      "default value":
      {
        "type": "unsigned int",
        "value": 943
      }
    },
    {
      "id": 28,
      "name": "iuli",
      "type":
      {
        "category": "fundamental type",
        "type": "unsigned long int"
      },
      "default value":
      {
        "type": "unsigned long int",
        "value": 1024
      }
    },
    {
      "id": 29,
      "name": "iulli",
      "type":
      {
        "category": "fundamental type",
        "type": "unsigned long long int"
      },
      "default value":
      {
        "type": "unsigned long long int",
        "value": 50000
      }
    }
  ]
}
